README.md

```diff
@@ -1,121 +1,115 @@
+* [You can use the utility functions such as `to_time_series_dataset`](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.utils.html#module-tslearn.utils)
+* [You can convert from other popular time series toolkits in Python](https://tslearn.readthedocs.io/en/latest/integration_other_software.html)
+* [You can load any of the UCR datasets in the required format.](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.datasets.html#module-tslearn.datasets)
+* [You can generate synthetic data using the `generators` module](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.generators.html#module-tslearn.generators)
+
+It should further be noted that tslearn [supports variable-length timeseries](https://tslearn.readthedocs.io/en/latest/variablelength.html).
+
+```python3
+>>> from tslearn.utils import to_time_series_dataset
+>>> my_first_time_series = [1, 3, 4, 2]
+>>> my_second_time_series = [1, 2, 4, 2]
+>>> my_third_time_series = [1, 2, 4, 2, 2]
+>>> X = to_time_series_dataset([my_first_time_series,
+                                my_second_time_series,
+                                my_third_time_series])
+>>> y = [0, 1, 1]
 ```
 
-## Using PyPI
+## 2. Data preprocessing and transformations
+Optionally, tslearn has several utilities to preprocess the data. In order to facilitate the convergence of different algorithms, you can [normalize time series](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.preprocessing.html#module-tslearn.preprocessing). Alternatively, in order to speed up training times, one can [resample](https://tslearn.readthedocs.io/en/latest/gen_modules/preprocessing/tslearn.preprocessing.TimeSeriesResampler.html#tslearn.preprocessing.TimeSeriesResampler) the data or apply a [piece-wise transformation](https://tslearn.readthedocs.io/en/latest/gen_modules/tslearn.piecewise.html#module-tslearn.piecewise), such as SAX.
+
+```python3
+>>> from tslearn.preprocessing import TimeSeriesScalerMinMax
+>>> from tslearn.piecewise import SymbolicAggregateApproximation
+>>> X_norm = TimeSeriesScalerMinMax().fit_transform(X)
+>>> X_sax = SymbolicAggregateApproximation(4, 10).fit_transform(X_norm)
+>>> print(X_sax)
+[[[5] [7] [8] [6]] 
+ [[5] [6] [8] [6]]
+ [[5] [6] [8] [6]]]
+```
 
-### Pre-requisites
+## 3. Training a model
 
-When using PyPI, C++ build tools should be available to perform installation.
+After getting the data in the right format, a model can be trained. Depending on the use case, tslearn supports different tasks: classification, clustering and regression. For an extensive overview of possibilities, check out our [gallery of examples](https://tslearn.readthedocs.io/en/latest/auto_examples/index.html).
 
-Using `pip` should also work fine:
-```bash
-python -m pip install tslearn
+```python3
+>>> from tslearn.shapelets import ShapeletModel
+>>> clf = ShapeletModel({3: 1})
+>>> clf.fit(X_sax, y)
```

Comment:
I would never try to run a `Shapelet` model on quantized time series representation such as SAX. Maybe fitting the model on `X_norm` would make more sense.

Reply:
That's true, that does not make that much sense indeed... I'll change that!